test/e2e/infra/secondary_network/secondary_network_prerequisite.sh

```diff
@@ -0,0 +1,360 @@
+
+_usage="Usage: $0 [--configure-prerequisites] [--deconfigure-prerequisites]
+                  [--base-interface <interface name>] [--sriov-vf-count <count>] [--host_name]
+
+Deploy Antrea CNI and its native secondary network prerequisites on an existing/running K8s cluster control node.
+
+        --configure-prerequisites     Deploy all the pre-requisite plugins and CRDs for secondary network configuration.
+        --deconfigure-prerequisites   Remove all the pre-requisite plugins and CRDs deployed for secondary network configuration.
+	--base-interface         Network interface to create VFs.
+        --sriov-vf-count         Number of virtual functions to be created on the base network interface.
+        --host-name            	 K8s cluster control node's host name or IP address."
+
+_ssh_config_info="Prerequisite: Kubernetes(K8s) cluster is up and running at the local or remote server.
+
+Follow these steps to configure SSH information to access the kubernetes cluster control node.
+
+1) Update remote/local cluster control node information at your /home/<user>/.ssh/config
+Example:
+Host <Control-Plane-Node>
+    HostName <Control-Plane-IP>
+    Port 22
+    user ubuntu
+    IdentityFile /home/ubuntu/.ssh/id_rsa
+
+2) Add remote/local cluster control node's public key (id_rsa.pub) to /home/<user>/.ssh/authorized_keys
+
+3) Set PubkeyAuthentication at sshd_config to yes at your remote/local cluster control node
+
+4) Esure SSH to the remote/local node works after the above configuration (ssh <control node/host name>)"
+
+function echoerr {
+        echo >&2 "$@"
+}
+
+function print_usage {
+	echoerr "$_usage"
+}
+
+function print_ssh_config_info {
+	echoerr "$_ssh_config_info"
+}
+
+function print_help {
+	echoerr "Try '$0 --help' for more information."
+}
+
+export URL_ANTREA_CONFIG_MAP='https://raw.githubusercontent.com/k8snetworkplumbingwg/sriov-network-device-plugin/master/deployments/configMap.yaml'
+export URL_SRIOV_DP_DAEMONSET='https://raw.githubusercontent.com/k8snetworkplumbingwg/sriov-network-device-plugin/master/deployments/k8s-v1.16/sriovdp-daemonset.yaml'
+export URL_WHEREABOUTS_IP_POOLS='https://raw.githubusercontent.com/k8snetworkplumbingwg/whereabouts/master/doc/crds/whereabouts.cni.cncf.io_ippools.yaml'
+export URL_WHEREABOUTS_OVERLAPPING_IP_RANGE_RES='https://raw.githubusercontent.com/k8snetworkplumbingwg/whereabouts/master/doc/crds/whereabouts.cni.cncf.io_overlappingrangeipreservations.yaml'
+export YAML_CONFIG_MAP='configMap.yaml'
+export YAML_SRIOV_DP_DAEMONSET='sriovdp-daemonset.yaml'
+export YAML_ANTREA='antrea.yml'
+export YAML_WHEREABOUTS_IP_POOL='whereabouts.cni.cncf.io_ippools.yaml'
+export YAML_WHEREABOUTS_OVERLAPPING_IP_RANGE_RES='whereabouts.cni.cncf.io_overlappingrangeipreservations.yaml'
+export YAML_NET_ATTACH_DEF_CRD='network-attachment-definition-crd.yaml'
+export YAML_SEC_NET_CONFIG_FILE='secondary_network_configuration.yaml'
+export YAML_VIRTUAL_NET_INSTANCE_01='virtual-network-instance1.yaml'
```

Comment:
Where we get these files? And you still choose to use YAML and kubectl, but not Go client to create these CRs? Why?

Reply:
Yes, @arunvelayutham please add these files as part of this PR.

I don't have a strong preference between creating the NetworkAttachment CRs as part of this script or as part of the Go test cases. I think either solution is fine. If these are hardcoded files that won't change, having the script create them makes sense IMO.

If we want NetworkAttachment CRs to be more dynamic, with the ability to have different test cases use different NetworkAttachments, then I agree that creating them as part of the Go tests using a programmatic client makes more sense. It seems that the only thing that the test cases would need to know in order to generate the CRs is the "base interface" name.

Reply:
the Virtual-network-instance.yml is a fixed file which doesn't intend to change based on the interface type or server execution @jianjuns. Based on @antoninbas suggestion above, I checked in this yml file as part of the this PR.

Reply:
I still prefer Go code, which is much easier to make changes or add new cases (e.g. when we change to Antrea IPAM or support overlay network) for new features. But I am fine if you do like to start with the YAML approach.

Reply:
We use NewPodBuilder() for Pod creation based on the test parameters (part of the secondary_network_test.go ). But, since the virtual network CRDs are pretty much fixed and not changed, I kept it as a predefined YAML and apply as part of the seconary_network_preprequisite.sh (earlier it was part of the go test suite. But based on @antoninbas's suggestion/comments, we moved all the setup stuffs into this script).

I agree that choosing golang for virtual network instance CRD creation makes it easier when we move out of whereabouts to Antrea IPAM. But, would be helpful if we take it up as a next step (while we move on to Antrea IPAM) while we get the current secondary network test enabled asap.

Reply:
Ok. Works for me.

Reply:
Thanks @jianjuns